### PR TITLE
update to activesupport 4.2

### PR DESCRIPTION
**Pull request by Ch4s3**

### CODE REVIEW

**Comment by stve** [comment]
any progress on this fellas? In the process of upgrading an app and just realized this is a blocker for me. Thanks.

**Comment by Ch4s3** [comment]
@stve It's a blocker for my team as well, but I/we haven't had time to figure it out yet. The only issue is the Mysql aes search discussed above. If you have any thoughts, we can probably get it knocked out soon. (As long as the problem isn't with ActiveRecord or the query builder)

File: lib/crypt_keeper/provider/mysql_aes_new.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def decrypt(value)
       #
       # Returns an Enumerable
       def search(records, field, criteria)
-        records.where(field.to_sym => encrypt(criteria))
+        records.select { |record| record[field] == criteria }
```
Comment (dwilkie) [inline lib/crypt_keeper/provider/mysql_aes_new.rb]: https://github.com/Ch4s3/crypt_keeper/pull/1 fixes the issue

**Comment by dwilkie** [comment]
https://github.com/Ch4s3/crypt_keeper/pull/1

**Comment by Ch4s3** [comment]
@jmazzi It looks like @dwilkie sorted out the MySql issue. It should be ready to go now.

**Comment by jmazzi** [comment]
@Ch4s3 @dwilkie awesome, did you want to open up a PR?

**Comment by Ch4s3** [comment]
@jmazzi This ( #94 ) is the pull request for the update. I merged @dwilkie 's  into support_rails_4_2 branch on Ch4s3/crypt_keeper.

**Comment by jmazzi** [comment]
@Ch4s3 nice! can you squash it into one commit and document the change? I will release it right after merging.

**Comment by Ch4s3** [comment]
@jmazzi  Its squashed and I added a bit to the Readme that says tested against 4.2, I wasn't sure what/where else to document the changes. I also added MRI 2.2 to the travis matrix for AR 4.2, since I feel like most people using 4.2 will want to use 2.2.

File: crypt_keeper.gemspec [hunk 2]
```diff
@@ -16,8 +16,8 @@ Gem::Specification.new do |gem|
   gem.require_paths = ["lib"]
   gem.version       = CryptKeeper::VERSION
 
-  gem.add_runtime_dependency 'activerecord',  '>= 3.1', '< 4.2'
-  gem.add_runtime_dependency 'activesupport', '>= 3.1', '< 4.2'
+  gem.add_runtime_dependency 'activerecord',  '>= 3.1', '<= 4.2'
+  gem.add_runtime_dependency 'activesupport', '>= 3.1', '<= 4.2'
```
Comment (jmazzi) [inline crypt_keeper.gemspec]: @Ch4s3 is there a specific reason you went with `<=` over `< 4.3`? If not, I'm going to change that in master after merging.

**Comment by jmazzi** [comment]
This fix was released as 0.19.0

File: crypt_keeper.gemspec (hunk 2 above)
Comment (Ch4s3) [inline crypt_keeper.gemspec]: No specific reason, but I'm not sure if 4.3 will be a thing or if they're jumping straight to 5.0, I haven't been following AR/Rails too closely lately.

**Pull request merged**